libraries/Filter/LowPassFilter2p.cpp

```diff
@@ -17,13 +17,12 @@ DigitalBiquadFilter<T>::DigitalBiquadFilter() {
 
 template <class T>
 T DigitalBiquadFilter<T>::apply(const T &sample, const struct biquad_params &params) {
-    if(is_zero(params.cutoff_freq) || is_zero(params.sample_freq)) {
+    if(!is_positive(params.cutoff_freq) || !is_positive(params.sample_freq)) {
```

Comment:
This changes to `!is_positive` to keep the same check as we do here:

https://github.com/ArduPilot/ardupilot/blob/f72bfcc373b0a61f327cccb0dc5812ecee17db8c/libraries/Filter/LowPassFilter2p.cpp#L54C1-L54C1

For example if a user was to set the cutoff to -1 for disable the  `compute_params` function would not update so you would be left with the configutation for the old cutoff and the filter would still work fine because -1 is not zero.